feat

zurabreon opened the pull request:
Выполнил техническое задание

zurabreon committed:
feat

Выполнил техническое задание

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+
+	app.post("/hook", (req, res) => {
+		
+		var contactId = req.body.contacts.add[0].id;
```
var лучше не использовать, заменить на const у всех переменных, которых возможно, у остальных на let

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+		
+		var contactId = req.body.contacts.add[0].id;
+		var customerAge;
+		var birthdayId = 403675;
```
такие переменные надо выносить в константы и нейминг таких("глобальных") констант в SNAKE_CASE капсом

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+			{
+				if (response.custom_fields_values[i].field_id === birthdayId)
+				{
+					const birthday = new Date(response.custom_fields_values[i].values[0].value * 1000);
```
тоже в константы

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+							customerAge--;
+						}
+					}
+					var customerAgeObject = {
```
const вместо var

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+
+api.getAccessToken().then(() => {
```
лучше использовать await вместо then, так код становится более читабельным, если нет необходимости именно в then. И в текущей реализации если в получении токена вылетит ошибка сервер просто не запустится

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+						}
+					}
+					var customerAgeObject = {
+						field_id: 410341,
```
откуда это число?

zurabreon replied:
Это id для поля возраста, которое нужно было создать по заданию

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+
+
+		api.getContact(contactId).then(response => {
+			for (i = 0; i < response.custom_fields_values.length; i++)
```
в utils.js есть функция получения значения кастомного поля и создание объекта кастомного поля для обновления

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,83 @@
+				{
+					const birthday = new Date(response.custom_fields_values[i].values[0].value * 1000);
+
+					if (currentData.getFullYear() - birthday.getFullYear() > 0) {
```
лучше сразу создать переменные с годом, месяцем, днем

zurabreon committed:
refactor

Избавился от цикла, воспользовавшись готовым методом из utils, корректировал код

zurabreon committed:
refactor

Корректировал код

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,85 @@
+		
+		const contactId = req.body.contacts.add[0].id;
+
+		const birthdayId = 467121; //id поля "День рождения"
```
оба эти id тоже надо вынести они же постоянны в данной задаче

zurabreon replied:
Переделал

psaraev1reon commented on index.js:
```diff
@@ -0,0 +1,85 @@
+		var customerAge;
+		
+
+		api.getContact(contactId).then(response => {
```
от then тоже можно избавиться

zurabreon replied:
Избавился

zurabreon committed:
refactor

aivanov-reon commented on index.js:
```diff
@@ -0,0 +1,80 @@
+
+const MILLISECONDS_PER_SECOND = 1000;
+
+const birthdayId = 467121; //id поля "День рождения"
```
Назвать как константу

zurabreon replied:
Переделал

aivanov-reon commented on index.js:
```diff
@@ -0,0 +1,80 @@
+			}
+		}
+
+		const updatedContactField = utils.makeField(ageId, customerAge)
```
Добавить проверку на undefined

zurabreon replied:
Сделал проверку

aivanov-reon commented on index.js:
```diff
@@ -0,0 +1,80 @@
+
+	app.post("/hook", async (req, res) => {
+		
+		const currentYear = new Date().getFullYear();
```
1. Вынести в отдельную функцию по расчету возраста
2. Установить пакет jest
3. Разобраться с пакетом jest
4. Написать тесты на функцию расчета возраста 5-6 проходок

zurabreon replied:
Вынес и написал тесты

aivanov-reon reviewed (commented):
Комментарии

zurabreon committed:
feat

Вынес функцию расчета отдельно, написал тесты для данной функции с помощью jest

aivanov-reon commented on index.js:
```diff
@@ -6,75 +6,48 @@ const express = require("express");
-		res.send("OK");
-		
-	});
+	} else {
```
if (updatedContactField) {
		const updateContactValues = {
			id: contact.id,
			custom_fields_values: [updatedContactField]

		};

		await api.updateContacts(updateContactValues);
		res.status(200).send({message: 'Ok'});
		return;
	}

	logger.error('Field is not defined', updatedContactField);

	res.status(400).send({message: 'bad request'});

zurabreon replied:
Переделал проверку на undefined. Сделал корректировки в calculationAge. Перепроверил условие

aivanov-reon reviewed (commented):
Коммент

zurabreon committed:
refactor

zurabreon committed:
refactor

aivanov-reon commented:
ОК
